packages/toolpad-app/src/toolpad/AppEditor/utils/deleteNode.ts

```diff
@@ -0,0 +1,31 @@
+import { NodeId } from '@mui/toolpad-core';
+import * as appDom from '../../../appDom';
+import { DomView } from '../../../utils/domView';
+import { AppStateApi } from '../../AppState';
+import deleteOrphanedLayoutNodes from './deleteOrphanedLayoutNodes';
+import normalizePageRowColumnSizes from './normalizePageRowColumnSizes';
+
+export default function deleteNode(
+  nodeId: NodeId,
+  pageNode: appDom.PageNode | null,
+  currentView: DomView,
+  appStateApi: AppStateApi,
+  dom: appDom.AppDom,
+) {
+  appStateApi.update(
+    (draft) => {
+      const toRemove = appDom.getNode(draft, nodeId);
```

Comment:
What if we kept just the part inside `appStateApi.update` as this reusable function, and so we didn't need to pass `appStateApi` and `currentView` as arguments, it would just be a function that transforms the DOM? 
I feel like it would be clearer that way and depend less on context, even if we had to repeat the other parts when we called the function.